### ISSUE TITLE
External ruleset is not found when specified in settings.json

### ISSUE THREAD

**Issue by CarloAxians** [HodorNV/ALOps#712]
**Describe the bug**
We have specified an external reference to a ruleset in the settings.json:

```
"al.ruleSetPath": "http://172.31.1.16:8080/ruleset/axians.ruleset.json",
"al.enableExternalRulesets": true,
```

In VSCode this ruleset can be found and the compilation is succesfull.
In Azure Devops/AlOps the file cannot be found.

**the used yaml**

```
 ## Compile App
  - task: ALOpsAppCompiler@2
    displayName: 'Compile App Extension ${{ alsourcepath }}'
    condition: succeeded()
    inputs:
      artifacttype: ${{parameters.artifact_type}}
      artifactcountry: ${{parameters.artifact_country}}
      artifactversion: ${{parameters.artifact_version}}
      versionselect: ${{parameters.version_select}}
      alsourcepath: ${{ alsourcepath }}
      appversiontemplate: '?.?.[yyyyMMdd].*'
      ${{ if contains(alsourcepath, 'test')}}:
        appfilenametemplate: '%APP_PUBLISHER%_%APP_NAME%_%APP_VERSION%_TEST.app'
      ${{ if not(contains(alsourcepath, 'test'))}}:
        appfilenametemplate: '%APP_PUBLISHER%_%APP_NAME%_%APP_VERSION%_APP.app'
      failonwarnings: $(failonwarnings)
      publishartifact: false
      alternativevsixurl: 'Latest'
      accept_insider_eula: true
```

**the output**
```
*** Using Ruleset: [http://172.31.1.16:8080/ruleset/axians.ruleset.json]
*** No AL-RuleSet json found.
```

**Expected behavior**
If we change the yaml file and add the ruleset to the ALOpsAppCompiler task it works fine:

```
 ## Compile App
  - task: ALOpsAppCompiler@2
    displayName: 'Compile App Extension ${{ alsourcepath }}'
    condition: succeeded()
    inputs:
      artifacttype: ${{parameters.artifact_type}}
      artifactcountry: ${{parameters.artifact_country}}
      artifactversion: ${{parameters.artifact_version}}
      versionselect: ${{parameters.version_select}}
      alsourcepath: ${{ alsourcepath }}
      appversiontemplate: '?.?.[yyyyMMdd].*'
      ${{ if contains(alsourcepath, 'test')}}:
        appfilenametemplate: '%APP_PUBLISHER%_%APP_NAME%_%APP_VERSION%_TEST.app'
      ${{ if not(contains(alsourcepath, 'test'))}}:
        appfilenametemplate: '%APP_PUBLISHER%_%APP_NAME%_%APP_VERSION%_APP.app'
      failonwarnings: $(failonwarnings)
      publishartifact: false
      alternativevsixurl: 'Latest'
      accept_insider_eula: true
      ruleset: 'http://172.31.1.16:8080/ruleset/axians.ruleset.json'
      enable_external_rulesets: true
```

**the output**
```
*** Downloading Ruleset settings: [http://172.31.1.16:8080/ruleset/axians.ruleset.json]
Invoke-RestMethod -Method Get -Uri "[http://172.31.1.16:8080/ruleset/axians.ruleset.json"](http://172.31.1.16:8080/ruleset/axians.ruleset.json%22) -OutFile "E:\Agents\Dynamics04\_work\63\s\app\ruleset.json"
*** Ruleset download complete
*** Using Ruleset: [E:\Agents\Dynamics04\_work\63\s\app\ruleset.json]
```

This is the outcome what we would also expect when we specify the URI in the settings.json only.

**Comment by CarloAxians** [comment]
hi @waldo1001, any update of this issue?

**Comment by waldo1001** [comment]
We found it and have added it for next release.

**Comment by waldo1001** [comment]
Should have been fixed in 1.462.
